### ISSUE TITLE
oxfmt: Markdown formatting is not idempotent when a table's delimiter row is indented deeper than its header row

### ISSUE THREAD

**Issue by enorm** [oxc-project/oxc#25612]
## Summary

A GFM table whose delimiter row is indented more deeply than its header row takes **two `oxfmt` runs** to reach a fixpoint. The first run dedents the delimiter row but leaves the table unaligned; the second aligns it.

This matters because it breaks `--check`: a file the formatter has just written is still reported as unformatted, so a "format, then verify" pipeline fails on its own output.

**Prettier behaves identically** — I checked before filing, and the intermediate state is the same. So this may well be intended compatibility rather than a defect, and I have no strong opinion on which. Filing it because the `--check` interaction is easy to hit in CI and hard to diagnose, and because if oxfmt ever wants to diverge from Prettier on idempotency, this is a concrete case.

## Reproduction

`t.md` — two lines, no config file, no `.editorconfig`:

```markdown
| x | y |
        |---|---|
```

```console
$ oxfmt t.md    # pass 1
| x | y |
|---|---|

$ oxfmt t.md    # pass 2
| x   | y   |
| --- | --- |

$ oxfmt t.md    # pass 3 — stable from here
| x   | y   |
| --- | --- |
```

And the `--check` symptom, which is the part that actually bites:

```console
$ printf '| x | y |\n        |---|---|\n' > t.md
$ oxfmt t.md            # format it
$ oxfmt --check t.md
t.md (79ms)
Format issues found in above 1 files. Run without `--check` to fix.
```

## Prettier comparison

```console
$ printf '| x | y |\n        |---|---|\n' > p.md
$ npx prettier@3 --write p.md   # pass 1 → | x | y | / |---|---|
$ npx prettier@3 --write p.md   # pass 2 → | x   | y   | / | --- | --- |
$ npx prettier@3 --check p.md   # after one pass: "Code style issues found"
```

Same two-pass behaviour, same intermediate output.

## Versions

- oxfmt **0.63.0** (also reproduces on **0.62.0**)
- macOS 15 / darwin arm64, installed both via Homebrew and as an npm devDependency
- No config file required to reproduce; also reproduces with an `.oxfmtrc.json` and an `.editorconfig` present

## How I hit it

Adopting oxfmt for the Markdown in a repository. One document had a table written this way, so `oxfmt --check` failed in CI immediately after the formatter had written the file. The workaround is running the formatter twice in write mode, which is what our script does now.

Nesting is not required — I originally found this under a list item and assumed indentation context mattered, but bisecting reduced it to the two lines above at top level.

**Comment by leaysgur** [comment]
Thanks for the report.

Currently, Oxfmt simply delegates the formatting of Markdown files to Prettier, so unfortunately, there is nothing that we can do for now.

Please follow: https://github.com/prettier/prettier/issues/19847
